arkane/encorr/isodesmic.py

```diff
@@ -155,89 +176,359 @@ def calculate_target_thermo(self):
+
+
+class BondConstraint:
+    def __init__(self, atom1, atom2, bond_order):
+        self.atom1 = atom1
+        self.atom2 = atom2
+        self.bond_order = bond_order
+
+    def __eq__(self, other):
+        if isinstance(other, BondConstraint):
+            if self.bond_order == other.bond_order:
+                if (self.atom1 == other.atom1 and self.atom2 == other.atom2) or (
+                    self.atom1 == other.atom2 and self.atom2 == other.atom1
+                ):
+                    return True
+            return False
+
+        if isinstance(other, GenericConstraint):
+            return False
+
+        else:
+            raise NotImplementedError(
+                f"BondConstraint object has no __eq__ defined for other object of type "
+                f"{type(other)}"
+            )
+
+    def __repr__(self):
+        symbols = ["", "-", "=", "#"]
+        return f"{self.atom1}{symbols[self.bond_order]}{self.atom2}"
+
+
+class Connection:
+    def __init__(self, atom, bond_order):
+        self.atom = atom
+        self.bond_order = bond_order
+
+    def __eq__(self, other):
+        if isinstance(other, Connection):
+            if self.bond_order == other.bond_order:
+                if self.atom == other.atom:
+                    return True
+            return False
+
+        else:
+            raise NotImplementedError(
+                f"Connection object has no __eq__ defined for other object of type {type(other)}"
+            )
+
+    def __repr__(self):
+        symbols = ["", "-", "=", "#"]
+        return f"{symbols[self.bond_order]}{self.atom}"
+
+
+class GenericConstraint:
+    def __init__(self, constraint_str: str):
+        self.constraint_str = constraint_str
+
+    def __eq__(self, other):
+        if isinstance(other, BondConstraint):
+            return False
+        elif isinstance(other, GenericConstraint):
+            return self.constraint_str == other.constraint_str
+        else:
+            raise NotImplementedError(
+                f"GenericConstraint object has no __eq__ defined for other object of "
+                f"type {type(other)}"
+            )
+    def __repr__(self):
+        return self.constraint_str
+
+
+def bond_centric_constraints(
+    species: ErrorCancelingSpecies, constraint_class: str
+) -> List[BondConstraint]:
+    constraints = []
+    contraint_func = CONSTRAINT_CLASSES[constraint_class]
+    molecule = species.molecule
+
+    for bond in molecule.get_all_edges():
+        constraints.append(contraint_func(bond))
+
+    return constraints
+
+
+def _buerger_rc2(bond: Bond) -> BondConstraint:
+    atom1 = bond.atom1
+    atom2 = bond.atom2
+
+    if atom1.symbol > atom2.symbol:
+        atom1, atom2 = atom2, atom1
+
+    atom1 = AtomConstraint(label=atom1.symbol)
+    atom2 = AtomConstraint(label=atom2.symbol)
+
+    return BondConstraint(atom1=atom1, atom2=atom2, bond_order=int(bond.order))
```

Comment:
Thanks for the question. It is not for the aromatic bond. Single and double bonds are saved as float in `Bond` object, 

```
mol.atoms[0].bonds
{<Atom 'C'>: <Bond "1.0">, <Atom 'C'>: <Bond "2.0">, <Atom 'H'>: <Bond "1.0">}
```

and float would cause this part of code to fail:

```
def __repr__(self):
        symbols = ["", "-", "=", "#"]
        return f"{symbols[self.bond_order]}{self.atom}"
```